pkg/util/metrics/version_parser.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"fmt"
+	apimachineryversion "k8s.io/apimachinery/pkg/version"
+	"regexp"
+)
+
+const (
+	versionRegexpString = `^v(\d+\.\d+\.\d+)`
```

Comment:
I don't know enough about the `apimachineryversion.Info` to know if we can rely on this.

@kubernetes/sig-api-machinery-pr-reviews

Reply:
I think it is really just legacy that it is actually called `apimachineryversion.Info`. The relevant scripts are build/release related. 

cc/ @BenTheElder

Reply:
We can probably just move the struct out while we are doing this.

Reply:
cc/ @sttts (I just noticed #39943) and it sounds like he'd have context.